### ISSUE TITLE
Add "callable" type.

### ISSUE THREAD

**Issue by jvprat** [gvvaughan/typecheck#5]
Right now the "function" type accepts both a plain lua function and table with the __call metamethod, but I think there's no way to force a function type.
Do you think it would be okay to make "function" accept only the plain functions and add a "callable" type that accepts both?

**Comment by gvvaughan** [comment]
That's an interesting topic actually.  When I first added the typecheck functionality in stdlib's debug module, there was the concept of a callable already... and back then I had imagined that functors (tables with a __call metamethod) were first class citizens of the core Lua implementation.  As I've worked through splitting out the features of stdlib into separately released projects like this one, and added specs to check the corner cases, it turns out that this is not the case at all :(

So, yes, it does seem useful to be able to distinguish between a raw function and a functor.  I think the logical conclusion to that is not to have a 'callable' as such, but to introduce a 'functor' type, and annotate the idea of a callable as 'function|functor' (although if we find ourselves typing that a lot, a third 'callable' type might turn out to be a useful shortcut in addition).
